LibGAP.jl/src/julia_to_gap.jl

```diff
@@ -103,16 +99,14 @@ function julia_to_gap(obj::Dict{T,S}, recursive::Val{Recursive}=Val(false), recu
     if haskey(recursion_dict,obj)
         return recursion_dict[obj]
     end
-    nr_entries = obj.count
-    keys = Array{T,1}(undef,nr_entries)
-    entries = Array{S,1}(undef,nr_entries)
-    i = 1
-    for (x,y) in obj
-        keys[i] = x
-        entries[i] = y
-        i += 1
-    end
-    record = GAP.Globals.CreateRecFromKeyValuePairList(julia_to_gap(keys,Val(true)),julia_to_gap(entries, recursive, recursion_dict))
+
+    k = collect(T, keys(obj))
+    v = collect(S, values(obj))
+
+    k = julia_to_gap(k, Val(true))
+    v = julia_to_gap(v, recursive, recursion_dict)
+
+    record = GAP.Globals.CreateRecFromKeyValuePairList(k, v)
     ## FIXME: This is too late!
```

Comment:
Actually, we could resolve this `FIXME` by creating an empty record at the beginning of the conversion function and then fill it with the converted stuff.

Reply:
... instead of using `GAP.Globals.CreateRecFromKeyValuePairList` at all. Yes, thought the same. I'll update the PR accordingly. And then I should put in a test for this, too...